### PR TITLE
docs(README) Update AWS README for module refactor

### PR DESCRIPTION
**Pull request by scottford-lw**
Re-write of the AWS README to document usage of the new TF modules. 

Signed-off-by: Scott Ford <scott.ford@lacework.net>

### CODE REVIEW

File: aws/README.md [hunk 1]
```diff
@@ -1,36 +1,96 @@
...
-The AWS account ID that is using the cross account role
-This is the Lacework AWS caller account ID
-434813966438
-Under “Options” click in the box to enable “Require external ID (Best practice when a third party will assume this role)”
-Create an External ID
+# Lacework Terraform Provisioning for AWS
+Terraform modules that create AWS resources required to integrate AWS accounts with the Lacework Cloud Security Platform.
+
+## AWS Config and CloudTrail Integration Overview
+In order for Lacework to monitor AWS configuration and CloudTrail activity, the following must be configured your AWS account:
```
Comment (afiune) [inline aws/README.md:5]:
```suggestion
In order for Lacework to monitor AWS configuration and CloudTrail activity, the following must be configured in your AWS account: 
```

File: aws/README.md [hunk 2]
```diff
@@ -1,36 +1,96 @@
...
-Under “Options” click in the box to enable “Require external ID (Best practice when a third party will assume this role)”
-Create an External ID
+# Lacework Terraform Provisioning for AWS
+Terraform modules that create AWS resources required to integrate AWS accounts with the Lacework Cloud Security Platform.
+
+## AWS Config and CloudTrail Integration Overview
+In order for Lacework to monitor AWS configuration and CloudTrail activity, the following must be configured your AWS account: 
+
+#### Required Resources
+- **Cross Account IAM Role** - Delegate access to Lacework to monitor resource configurations within customer's AWS account. This role is used for both the Config and CloudTrail
```
Comment (afiune) [inline aws/README.md:8]:
```suggestion
- **Cross Account IAM Role** - Delegate access to Lacework to monitor resource configurations within the customer's AWS account. This role is used for both, Config and CloudTrail
```

File: aws/README.md [hunk 3]
```diff
@@ -1,36 +1,96 @@
...
+  - `SecurityAudit` Policy - AWS managed policy to allow Lacework to assess configuration. Policy is applied for configuration assessment only.
+  - Custom IAM Policy - Delegate access to Lacework to monitor CloudTrail Activity. Policy is attached to the IAM role when for CloudTrail is configured.
```
Comment (afiune) [inline aws/README.md:10]:
```suggestion
  - Custom IAM Policy - Delegate access to Lacework to monitor CloudTrail Activity. The policy is attached to the IAM role when CloudTrail is configured
```

File: aws/README.md [hunk 4]
```diff
@@ -1,36 +1,96 @@
...
-Create an External ID
+# Lacework Terraform Provisioning for AWS
+Terraform modules that create AWS resources required to integrate AWS accounts with the Lacework Cloud Security Platform.
+
+## AWS Config and CloudTrail Integration Overview
+In order for Lacework to monitor AWS configuration and CloudTrail activity, the following must be configured your AWS account: 
+
+#### Required Resources
+- **Cross Account IAM Role** - Delegate access to Lacework to monitor resource configurations within customer's AWS account. This role is used for both the Config and CloudTrail
+  - `SecurityAudit` Policy - AWS managed policy to allow Lacework to assess configuration. Policy is applied for configuration assessment only.
```
Comment (afiune) [inline aws/README.md:9]:
```suggestion
  - `SecurityAudit` Policy - AWS managed policy to allow Lacework to assess configuration. The policy is applied for configuration assessment only
```

File: aws/README.md [hunk 5]
```diff
@@ -1,36 +1,96 @@
...
+  - Custom IAM Policy - Delegate access to Lacework to monitor CloudTrail Activity. Policy is attached to the IAM role when for CloudTrail is configured.
+- **CloudTrail** - Enable a new CloudTrail or use an existing
+  - **S3 Bucket** - Used to store CloudTrail logs. Create a new S3 bucket, or use an existing
+  - **SNS Topic** - Used to send notifications when CloudTrail publishes new log files to the configured S3 bucket. Use existing or create a new SNS topic.
```
Comment (afiune) [inline aws/README.md:13]:
```suggestion
  - **SNS Topic** - Used to send notifications when CloudTrail publishes new log files to the configured S3 bucket. Use existing or create a new SNS topic
```

File: aws/README.md [hunk 6]
```diff
@@ -1,36 +1,96 @@
...
+  - **SQS Queue** - SQS queue subscribed to CloudTrail SNS topic for Lacework  
+- **Lacework AWS CFG Integration** - Configures AWS CFG integration between Lacework and customer AWS account
+- **Lacework AWS CT Integration** - Configures AWS CT integration between Lacework and customer AWS account
+
+## Requirements
+Before you begin the following must be configured on the workstation running Terraform
```
Comment (afiune) [inline aws/README.md:19]:
```suggestion
Before you begin the following must be configured on the workstation running Terraform:
```

File: aws/README.md [hunk 7]
```diff
@@ -1,36 +1,96 @@
...
+
+- [AWS API Access Key, Secret Access Key](https://aws.amazon.com/premiumsupport/knowledge-center/create-access-key/)
+- [Terraform](terraform.io/downloads.html) `0.12.x`
+- [Lacework API Key](https://support.lacework.com/hc/en-us/articles/360011403853-Generate-API-Access-Keys-and-Tokens)
+
+## Usage
+
+### Enable New CloudTrail Configuration
+This example enables a new CloudTrail, IAM Role for Lacework, and then configures both integrations with Lacework
```
Comment (afiune) [inline aws/README.md:28]:
```suggestion
This example enables a new CloudTrail, IAM Role for Lacework, and then configures both integrations with Lacework:
```

File: aws/README.md [hunk 8]
```diff
@@ -1,36 +1,96 @@
...
+  source                = "./modules/cloudtrail"
+  bucket_force_destroy  = true
+  use_existing_iam_role = true
+  iam_role_name         = module.aws_config.iam_role_name
+  iam_role_external_id  = module.aws_config.external_id
+}
+```
+
+### Integrate Existing CloudTrail with Lacework
+This example uses an existing CloudTrail, S3 bucket, and SNS topic passed as inputs to the module. The example creates the SQS queue and IAM Role for Lacework, and then configures both integrations with Lacework
```
Comment (afiune) [inline aws/README.md:49]:
```suggestion
This example uses an existing CloudTrail, S3 bucket, and SNS topic passed as inputs to the module. The example creates the SQS queue and IAM Role for Lacework, then configures both integrations with Lacework.
```

File: aws/README.md [hunk 9]
```diff
@@ -1,36 +1,96 @@
...
+| external_id_length | Length of External ID (max 1224) | `number` | 16 | no |
+| iam_role_external_id | External ID for IAM Role | `string` | "" | no |
+| iam_role_name |  The IAM role name | `string` | "lacework_iam_role" | no |
+| lacework_account_id | The Lacework AWS account that the IAM role will grant access | `string` | 434813966438 | no |
+| lacework_integration_name | The name of the integration in Lacework | `string` | TF config | no |
+| prefix | The prefix that will be use at the beginning of every generated resource | `string` | lacework-ct | no |
+| sns_topic_name | SNS topic name. Can be used when generating a new resource or when using an existing resource. | `string` | "" | no |
+| sqs_queue_name | SQS queue name. Can be used when generating a new resource or when using an existing resource. | `string` | "" | no |
+| use_existing_cloudtrail | Set this to true to use an existing cloudtrail | `bool` | `false` | no |
+| use_existing_iam_role | Set this to true to use an existing IAM role | `bool` | `false` | no |
```
Comment (afiune) [inline aws/README.md:88]: I would make sure to explain here that, if a user sets this to `true`, they must provide the `iam_role_name` and `iam_role_external_id`.

File: aws/README.md [hunk 10]
```diff
@@ -1,36 +1,96 @@
...
+| cloudtrail_name | Name of existing cloudtrail | `string` | "lacework-cloudtrail" | no |
+| external_id_length | Length of External ID (max 1224) | `number` | 16 | no |
+| iam_role_external_id | External ID for IAM Role | `string` | "" | no |
+| iam_role_name |  The IAM role name | `string` | "lacework_iam_role" | no |
+| lacework_account_id | The Lacework AWS account that the IAM role will grant access | `string` | 434813966438 | no |
+| lacework_integration_name | The name of the integration in Lacework | `string` | TF config | no |
+| prefix | The prefix that will be use at the beginning of every generated resource | `string` | lacework-ct | no |
+| sns_topic_name | SNS topic name. Can be used when generating a new resource or when using an existing resource. | `string` | "" | no |
+| sqs_queue_name | SQS queue name. Can be used when generating a new resource or when using an existing resource. | `string` | "" | no |
+| use_existing_cloudtrail | Set this to true to use an existing cloudtrail | `bool` | `false` | no |
```
Comment (afiune) [inline aws/README.md:87]: I would make sure to explain here that, if a user sets this to `true`, they must provide the `bucket_name` and `sns_topic_name`.

File: aws/README.md (hunk 9 above)
Comment (scottford-lw) [reply aws/README.md]: @afiune 👍

File: aws/README.md (hunk 10 above)
Comment (scottford-lw) [reply aws/README.md]: @afiune 👍

File: aws/README.md [hunk 11]
```diff
@@ -1,36 +1,96 @@
...
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| bucket_force_destroy | Force destroy bucket (Required when bucket not empty) | `bool` | false | no |
+| bucket_name | Name of S3 bucket | `string` | lacework- | no |
+| cloudtrail_name | Name of existing cloudtrail | `string` | "lacework-cloudtrail" | no |
+| external_id_length | Length of External ID (max 1224) | `number` | 16 | no |
+| iam_role_external_id | External ID for IAM Role | `string` | "" | no |
+| iam_role_name |  The IAM role name | `string` | "lacework_iam_role" | no |
+| lacework_account_id | The Lacework AWS account that the IAM role will grant access | `string` | 434813966438 | no |
+| lacework_integration_name | The name of the integration in Lacework | `string` | TF config | no |
```
Comment (afiune) [inline aws/README.md:83]: One last thing; This variable is available for both modules, it is the name of the integration for config and cloudtrail, they both have defaults.

Comment (scottford-lw) [reply aws/README.md]: 👍  💯

**Pull request merged**